internal/devices/create.go

```diff
@@ -97,6 +97,17 @@ func (c *Client) Create() *cobra.Command {
 			var facilityArgs []string
 
 			var request metal.ApiCreateDeviceRequest
+
+			var validBillingCycle *metal.DeviceCreateInputBillingCycle
+			var err error
+
+			if billingCycle != "" {
+				validBillingCycle, err = metal.NewDeviceCreateInputBillingCycleFromValue(billingCycle)
+				if err != nil {
+					return err
```

Comment:
This is returning the original error because that message is nicely descriptive: https://github.com/equinix-labs/metal-go/blob/main/metal/v1/model_device_create_input_billing_cycle.go#L52

The error message does make mention of the type `DeviceCreateInputBillingCycle`; it would be preferable to reference either the CLI flag name or the attribute name there, but IMO tweaking that is not worth the effort at the moment.